rust/hyperlane-base/src/metadata.rs

```diff
@@ -0,0 +1,10 @@
+use derive_new::new;
+use serde::{Deserialize, Serialize};
+
+/// Metadata about agent
+#[derive(Debug, Deserialize, Serialize, new)]
+pub struct AgentMetadata {
+    /// Latest git commit hash at the time when agent was built
+    /// If .git was not present at the time of build, field defaults to "VERGEN_IDEMPOTENT_OUTPUT"
```

Comment:
nit: feels like this info is most useful to know where the `git_sha` is created (i.e where `env!("VERGEN_GIT_SHA")` is called), wonder if a note about this at the `AgentMetadata::new` callsite would be useful, or maybe rolling that `env!("VERGEN_GIT_SHA")` bit into a non-derived `AgentMetadata::new()`

Reply:
I moved documentation comments around.

I would not move `env!("VERGEN_GIT_SHA")` inside the constructor of the struct instance since it reduces the flexibility. We may want to create an instance of the struct with a specific hash for testing purposes, for example.